snakemake/remote/GS.py

```diff
@@ -125,41 +148,29 @@ def size(self):
         else:
             return self._iofile.size_local
 
-    def download(self, retry_count=3):
+    @retry.Retry(predicate=google_cloud_retry_decorator)
+    def download(self):
         if not self.exists():
             return None
 
         # Create the directory for the intended file
         os.makedirs(os.path.dirname(self.local_file()), exist_ok=True)
-        checksums_match = False
-
-        # Use boolean to not stress filesystem on each loop
-        while not checksums_match:
-
-            # ideally we could calculate hash while streaming to file with provided function
-            # https://github.com/googleapis/python-storage/issues/29
-            with open(self.local_file(), "wb") as blob_file:
-                parser = Crc32cCalculator(blob_file)
-                self.blob.download_to_file(parser)
-            os.sync()
-
-            # Compute local hash and verify correct
-            if parser.hexdigest() != self.blob.crc32c:
-                os.remove(self.local_file())
-                retry_count -= 1
-            else:
-                checksums_match = True
-
-            # After three tries, likely not going to work
-            if retry_count == 0:
-                raise WorkflowError(
-                    "Failed to download {} from remote: checksums did not match {} times.".format(
-                        self.remote_file(), retry_count
-                    )
-                )
+
+        # ideally we could calculate hash while streaming to file with provided function
+        # https://github.com/googleapis/python-storage/issues/29
+        with open(self.local_file(), "wb") as blob_file:
+            parser = Crc32cCalculator(blob_file)
+            self.blob.download_to_file(parser)
+        os.sync()
+
+        # Compute local hash and verify correct
+        if parser.hexdigest() != self.blob.crc32c:
+            os.remove(self.local_file())
+            raise CheckSumMismatchException
```

Comment:
Please add a proper error message here, saying that checksum of file ... was wrong.

Reply:
Gotcha!